File: CODIGOS5/173.lua
function main()
    local numTestCases = tonumber(io.read())

    for i = 1, numTestCases do
        local inputNotes = io.read()
        local notes = {}
        for note in inputNotes:gmatch("%S+") do
            table.insert(notes, tonumber(note))
        end

        local chordName = determineChordName(notes)
        io.write(chordName, " ")
    end
end

function determineChordName(notes)
    local steps = {}
    for i = 2, #notes do
        local diff = (notes[i] - notes[i - 1] + 12) % 12
        table.insert(steps, diff)
    end

    if contains(steps, 3) and contains(steps, 4) then
        return "other"
    elseif contains(steps, 4) then
        return getChordName(notes[1], true)
    elseif contains(steps, 3) then
        return getChordName(notes[1], false)
    else
        return "other"
    end
end

function getChordName(rootNote, isMajor)
    local noteNames = {
        "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B"
    }
    local rootNoteName = noteNames[rootNote % 12 + 1]
    if isMajor then
        return rootNoteName .. "-major"
    end
    return rootNoteName .. "-minor"
end

function contains(arr, target)
    for _, num in ipairs(arr) do
        if num == target then
            return true
        end
    end
    return false
end

main()

File: CODIGOS5/173.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	var numTestCases int
	fmt.Scanln(&numTestCases)

	scanner := bufio.NewScanner(os.Stdin)
	for i := 0; i < numTestCases; i++ {
		scanner.Scan()
		inputNotes := strings.Fields(scanner.Text())
		notes := make([]int, len(inputNotes))
		for j, note := range inputNotes {
			notes[j], _ = strconv.Atoi(note)
		}

		chordName := determineChordName(notes)
		fmt.Printf("%s ", chordName)
	}
}

func determineChordName(notes []int) string {
	steps := make([]int, len(notes)-1)
	for i := 1; i < len(notes); i++ {
		diff := (notes[i] - notes[i-1] + 12) % 12
		steps[i-1] = diff
	}

	if contains(steps, 3) && contains(steps, 4) {
		return "other"
	} else if contains(steps, 4) {
		return getChordName(notes[0], true)
	} else if contains(steps, 3) {
		return getChordName(notes[0], false)
	} else {
		return "other"
	}
}

func getChordName(rootNote int, isMajor bool) string {
	noteNames := []string{"C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B"}
	rootNoteName := noteNames[rootNote%12]
	if isMajor {
		return fmt.Sprintf("%s-major", rootNoteName)
	}
	return fmt.Sprintf("%s-minor", rootNoteName)
}

func contains(arr []int, target int) bool {
	for _, num := range arr {
		if num == target {
			return true
		}
	}
	return false
}
